### PR TITLE
Delete contact from API

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: temba/api/views.py [hunk 1]
```diff
@@ -1117,6 +1134,24 @@ def post(self, request, format=None):
...
 
+    def delete(self, request, *args, **kwargs):
+        return self.destroy(request, *args, **kwargs)
+
+    def destroy(self, request, *args, **kwargs):
+        self.request = request
+        queryset = self.get_queryset()
+
+        # don't let users delete all their contacts by mistake
+        if not self.request.QUERY_PARAMS:
```
Comment (norkans7) [inline temba/api/views.py]: I think we have to check also that we have the right parameters here

Comment (rowanseymour) [inline temba/api/views.py]: Good call. Updating but also wondering if everyone agrees that the API shouldn't let people delete all contacts.

Comment (norkans7) [inline temba/api/views.py]: I even doubt if we have to allow deleting all contacts in a group, may be allow deleting specified urns and uuids only

**Comment by nicpottier** [comment]
Nice, looks good.

**Comment by rowanseymour** [comment]
Incidentally, users now sorta have a way to delete all contacts in an org...

```python
from temba import TembaClient
client = TembaClient("rapidpro.io", <APITOKEN>)
for c in client.get_contacts():
  client.delete_contact(c.uuid)
```

**Comment by nicpottier** [comment]
Well, with great power comes great responsibility..

**Pull request merged**